src/ui/public/index_patterns/index_patterns.js

```diff
@@ -41,6 +41,7 @@ export function IndexPatternsProvider(Notifier, Private) {
   self.cache = patternCache;
   self.getIds = getProvider('id');
   self.getTitles = getProvider('attributes.title');
+  self.getIndexPatterns = getProvider();
```

Comment:
I'm actually a little confused by all of these names.  :)

We have `IndexPatternsProvider` `IndexPatternProvider` and `IndexPatternsGetProvider`.   `IndexPatternProvider` provides the class IndexPattern.  But whats the difference between `IndexPatterns` and `IndexPatternsGet`?  The only thing I can relate it to is `SavedObject` and `SavedObjectLoader` but which one is the loader here, and what's the point of the other one?

I think the name `getProvider` is also confusing.  We generally drop the `provider` part when we wrap it in Private, e.g.:
```
  const IndexPattern = Private(IndexPatternProvider);
  const patternCache = Private(IndexPatternsPatternCacheProvider)
  const getProvider = Private(IndexPatternsGetProvider); // Different from the rest
```

so I think to match the others, but give more context than just `get`, it should be named `indexPatternsGet`, though that's not a great name in itself.

I like @chrisronline's suggestion, and agree it pairs well with the base `getIndexPatterns`, and I'm generally in favor of more verbosity.  Not super necessary but to break the tie, I'll jump in his camp.

Is there a reason not to have getIds use getIndexPatterns and map to an array of ids?  Or is there something more efficient about using the getIds function returned from the getProvider directly?

@nreese - what are you suggesting be renamed `getSavedObjects`?  `getProvider`? or `getIndexPatterns`?  What is returning saved objects here?  This is confusing me even more.  getIds returns IndexPattern ids, right? not saved object ids?

I know some of my name nitpicks are for existing code, so if those names are really baked into the code base, or you don't feel like a big refactor, or disagree with my sentiments, I'll defer to leave as is.